src/v/raft/consensus_utils.cc

```diff
@@ -451,4 +455,84 @@ ss::future<> move_persistent_state(
     });
 }
 
+ss::future<> create_offset_translator_state_for_pre_existing_partition(
+  storage::api& api,
+  const storage::ntp_config& ntp_cfg,
+  raft::group_id group,
+  model::offset min_rp_offset,
+  model::offset max_rp_offset) {
+    // Prepare offset_translator state in kvstore
+    storage::offset_translator_state ot_state(
+      ntp_cfg.ntp(), raft::details::prev_offset(min_rp_offset), 0);
+    co_await api.kvs().put(
+      storage::kvstore::key_space::offset_translator,
+      raft::offset_translator::kvstore_offsetmap_key(group),
+      ot_state.serialize_map());
+    co_await api.kvs().put(
+      storage::kvstore::key_space::offset_translator,
+      raft::offset_translator::kvstore_highest_known_offset_key(group),
+      reflection::to_iobuf(raft::details::prev_offset(max_rp_offset)));
+}
+
+ss::future<> create_raft_state_for_pre_existing_partition(
+  storage::api& api,
+  const storage::ntp_config& ntp_cfg,
+  raft::group_id group,
+  model::offset min_rp_offset,
+  model::offset max_rp_offset,
+  model::term_id last_included_term,
+  std::vector<model::broker> initial_nodes) {
+    // Prepare Raft state in kvstore
+    auto key = raft::details::serialize_group_key(
+      group, raft::metadata_key::config_latest_known_offset);
+    co_await api.kvs().put(
+      storage::kvstore::key_space::consensus,
+      key,
+      reflection::to_iobuf(max_rp_offset));
+    co_await api.kvs().put(
```

Comment:
Nit: contrary to what the comment says, this is not raft state, this is disk log state. So I would expect it to be placed a bit separately.